.github/workflows/docs_build_and_deploy.yml

```diff
@@ -12,23 +12,35 @@ on:
     tags:
       - '*'
   pull_request:
+  merge_group:
   workflow_dispatch:
 
 jobs:
+  linting:
+    # scheduled workflows should not run on forks
+    if: (${{ github.event_name == 'schedule' }} && ${{ github.repository_owner == 'neuroinformatics-unit' }} && ${{ github.ref == 'refs/heads/main' }}) || (${{ github.event_name != 'schedule' }})
+    runs-on: ubuntu-latest
+    steps:
+      - uses: neuroinformatics-unit/actions/lint@v2
+
   build_sphinx_docs:
     name: Build Sphinx Docs
     runs-on: ubuntu-latest
     steps:
-      - uses: neuroinformatics-unit/actions/build_sphinx_docs@v2
+      - uses: neuroinformatics-unit/actions/build_sphinx_docs@main
+        with:
+          python-version: 3.11
+          use-make: true
 
   deploy_sphinx_docs:
     name: Deploy Sphinx Docs
     needs: build_sphinx_docs
     permissions:
       contents: write
-    if: github.event_name == 'push' && github.ref_type == 'tag'
+    if: (github.event_name == 'push' && github.ref_type == 'tag') || github.event_name == 'workflow_dispatch'
```

Comment:
If I understand correctly, if you are testing e.g. on a pull request and the tests fail, but you manually re-run the the failed ones all for some reason (e.g. to check it was not some stochastic thing in your tests that caused the fail), it will rebuild the docs?

Reply:
I think it is there to manually trigger the docs building workflow in case it fails. What you propose is a case for which it can happen.

Reply:
I think this is very useful, I sometimes wish for it in datashuttle too. But I wonder if there is another level of protection that can be given, to stop it being accidentally triggered in a case like the above. Or, maybe it only happens if you request the entire workflow manually (rather than retry some tests that failed on a PR). 

Actually, I think that is the case, because it says [here](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/manually-running-a-workflow) you must be on a default branch. So I think probably okay as is!